app/src/main/java/ingsw/group1/findmyphone/event/SMSLogEvent.java

```diff
@@ -0,0 +1,199 @@
+package ingsw.group1.findmyphone.event;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+import java.util.Objects;
+
+import ingsw.group1.findmyphone.contacts.SMSContact;
+import ingsw.group1.findmyphone.location.GeoPosition;
+import ingsw.group1.msglibrary.SMSPeer;
+
+/**
+ * Class defining a {@link LoggableEvent} that uses instances of {@link SMSContact}.
+ *
+ * @author Riccardo De Zen.
+ */
+public final class SMSLogEvent implements LoggableEvent<SMSContact> {
```

Comment:
Why final?

Reply:
A library I used to check the equals and hashcode methods suggested it, if a subclass exists it's not accepted by the equals method, I have to further check this

Reply:
I analyzed the issue and the class didn't need to be final. I put it back to normal